collect/central_collector.go

```diff
@@ -119,18 +153,235 @@ func (c *CentralCollector) collect() {
 
 }
 
-func (c *CentralCollector) processSpan(sp *types.Span) {
-	// add the span to the cache
-	trace := c.SpanCache.Get(sp.TraceID)
-	if trace == nil {
-		err := c.SpanCache.Set(sp)
-		if err != nil {
-			c.Logger.Error().Logf("error adding span with trace ID %s to cache: %s", sp.TraceID, err)
-			return
+func (c *CentralCollector) processTraces() error {
+	for {
+		select {
+		case <-c.done:
+			return nil
+		default:
+			ids := c.SpanCache.GetTraceIDs(processTracesBatchSize)
+			if len(ids) == 0 {
+				continue
```

Comment:
Let's create a metric histogram for the count returned from GetTraceIDs -- it will help in tuning the batch size.